plugins/aea-cli-ipfs/aea_cli_ipfs/ipfs_utils.py

```diff
@@ -324,29 +324,28 @@ def download(
         :return: downloaded path
         """
 
-        def move_to_target_dir(download_tmp_path) -> str:
+        def move_to_target_dir(download_tmp_path: Path) -> str:
             """Move downloaded content to target directory"""
             if download_tmp_path.is_file():
-                # file is ok!
+                # Downloaded object is a file and can be moved easily
                 shutil.move(str(download_tmp_path), str(target_dir))
                 return str(target_dir)
 
             if fix_path:
                 # identify name of package
                 paths = list(download_tmp_path.glob("*"))
-                assumption_is_valid = len(paths) == 1 and paths[0].is_dir()
-                if not assumption_is_valid:  # pragma: no cover
-                    error_msg = f"Expected a single directory, found: {paths}"
-                    raise DownloadError(error_msg)
-                package_path = paths.pop()
+                if len(paths) == 1:
+                    (package_path,) = paths
+                else:
+                    package_path = download_tmp_path
```

Comment:
Continuing from ..., this will fix the issue but we need to make this method more robust in terms of working with different types of download objects

Reply:
@angrybayblade @Karrenbelt 
i think it's better to get rid of package terms in ipfs download tool.
the main purpose is to download:
* a single file
* a directory

and doesnt matter what is actually inside.

wrapped single file is a directory actually. with one file inside and it's filename
a directory downloaded by hash stored (default ipfs cli behaviour) as  `<target dir><hash>/<dir content>` sometimes we want to have `<target dir>/<dir contet>` without dir with name `<hash>`

quite simple logic. content related operations should be done on the upper level.

my conclusion:

make ipfstool as simple as possible (download dir, file, retries)
content related things should be implemented somewhere else.

Reply:
Please open an issue for this so when we're working https://github.com/valory-xyz/open-aea/issues/519 we work on this afterwards